app/models/miq_ae_yaml_import.rb

```diff
@@ -15,10 +15,17 @@ def initialize(domain, options)
   def import
     if @options.key?('import_dir') && !File.directory?(@options['import_dir'])
       raise MiqAeException::DirectoryNotFound, "Directory [#{@options['import_dir']}] not found"
+    elsif !User.current_user.nil? && domain_locked?
+      raise MiqAeException::DomainNotAccessible, 'locked domain'
     end
     start_import(@options['preview'], @domain_name)
   end
 
+  def domain_locked?
+    dom = domain_files('*').first.split('/').first
+    (@options['import_as'] && MiqAeDomain.find_by(:name => @options['import_as'])&.contents_locked?) || (dom && MiqAeDomain.find_by(:name => dom)&.contents_locked?)
```

Comment:
@pkomanek As discussed it would be good to move this line into a separate method with a descriptive name, otherwise it is difficult to parse at first glance.